driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/MongoIterableSubscription.java

```diff
@@ -81,6 +81,9 @@ void requestMoreData() {
         }
```

Comment:
I think you can just add `&& !batchCursor.isClosed()` to the must read check.

How to test this? I think its a race condition, unsubscribe and call request at the same time it could hit.  So I'm not sure it can be tested reliably.

Reply:
I have made the code change. In this case, we do not throw an exception when the batch cursor is closed, thus failing silently. This is a different approach than in `ClientSessionHelper` where we pass an exception to the callback if the cluster is closed.

Reply:
Failing silently sounds scary.  Generally we want to let the application know about error conditions.  Can you explain why this is the correct approach?

Reply:
I was not suggesting that this is the correct approach, only providing a choice. I have added a call to `onError` to let the application know.

Reply:
I think returning nothing to the callback here is the correct approach.  We can't call onError after we've unsubscribed to the Publisher.  The original fix was correct if batchCursor is closed mustRead should be false.

Reply:
I have reverted this change to silently fail. My mistake.

Reply:
requestMoreData can be a noop - so I think this is the correct approach.